lib/internal/http2/core.js

```diff
@@ -442,7 +445,7 @@ function setupHandle(session, socket, type, options, settings) {
     handle.onstreamclose = onSessionStreamClose;
     handle.onerror = onSessionError;
     handle.onread = onSessionRead;
-    handle.ongetpadding = onSelectPadding;
+    handle.ongetpadding = onSelectPadding(options.selectPadding);
```

Comment:
we are allocating a closure for no reason if `options.selectPadding`  is undefined, as I expect it to be the "base case".